Shared/ViewModels/ActiveSessionsViewModel.swift

```diff
@@ -0,0 +1,138 @@
+//
+// Swiftfin is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2024 Jellyfin & Jellyfin Contributors
+//
+
+import Combine
+import Foundation
+import JellyfinAPI
+import OrderedCollections
+import SwiftUI
+
+final class ActiveSessionsViewModel: ViewModel, Stateful {
+
+    // MARK: Action
+
+    enum Action: Equatable {
+        case error(JellyfinAPIError)
+        case loadSessions
+        case refreshSessions
+    }
+
+    // MARK: State
+
+    enum State: Hashable {
+        case initial
+        case loading
+        case loaded
+        case refreshing
+        case error(JellyfinAPIError)
+    }
+
+    @Published
+    var sessions: [SessionInfo] = []
+
+    @Published
+    final var state: State = .initial
+
+    @Published
+    final var lastAction: Action?
+
+    private var timer: Timer?
+
+    // MARK: Initialization
+
+    override init() {
+        super.init()
+        Task { @MainActor in
```

Comment:
Don't start loading in an `init`. Instead, have the view send the `loadSessions` in a `.onFirstAppear`.